logprep/ng/abc/event.py

```diff
@@ -72,10 +72,13 @@ def __init__(
         True
         """
         self._state: EventState = EventState()
-        if state is not None:
-            if state not in list(EventStateType):
-                raise TypeError("state must be an instance of EventStateType or None")
+
+        if isinstance(state, EventState):
+            self._state = state
+        elif state is not None and state in list(EventStateType):
             self._state.current_state = state
```

Comment:
As discussed, please rename current_state to _current_state in EventState class and adapt this line to:
```suggestion
            self._state._current_state = state
```

Reply:
as seen here: https://github.com/fkie-cad/Logprep/pull/857/commits/cc8dd3311137fa1c61722cd688a4b9b3073a082b this change breaks everything surprisingly. I  suggest to make this change not now, because it blocks too much now.